src/scss/grommet-core/_objects.layer.scss

```diff
@@ -143,6 +143,10 @@
   }
 
   &--hidden {
+
+    left: -10000px;
```

Comment:
The side that the Layer is hidden on should be tied to the layer direction, not all layers should be hidden to the left.